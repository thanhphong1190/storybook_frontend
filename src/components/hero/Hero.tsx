import * as S from './Hero.styles'
import { TextProps, ButtonProps } from '../../services/api/types'

export interface HerorTextProps extends TextProps {
  spaceTop?: number
  spaceBottom?: number
}
export interface HeroProps {
  bgImgURL?: string
  bgColor?: 'orange' | 'blue' | 'dark-blue' | 'dodger-blue' | 'white' | 'grey'
  contentPosition?: 'center' | 'left' | 'right'
  contentWidth?: 'full' | '1/2' | '2/3' | '3/4' | '4/5'
  texts?: HerorTextProps[]
  buttons?: ButtonProps[]
  overlay?: boolean
}

export const Hero = (props: HeroProps) => {
  const {
    bgImgURL,
    bgColor,
    contentPosition,
    overlay,
    texts,
    buttons,
    contentWidth
  } = props
  return (
    <S.Section bgImgURL={bgImgURL} bgColor={bgColor}>
      <S.OverlayWrapper className={overlay ? `bg-opacity-50 bg-black` : ''}>
        <S.Container contentPosition={contentPosition}>
          <S.ContentWrapper width={contentWidth}>
            {texts &&
              texts.map(textProp => (
                <S.HeroText {...textProp}>{textProp.text}</S.HeroText>
              ))}
            <S.ButtonGroup contentPosition={contentPosition}>
              {buttons &&
                buttons.map(but => (
                  <S.HeroButtonLink {...but}>{but.label}</S.HeroButtonLink>
                ))}
            </S.ButtonGroup>
          </S.ContentWrapper>
        </S.Container>
      </S.OverlayWrapper>
    </S.Section>
  )
}

export default Hero
