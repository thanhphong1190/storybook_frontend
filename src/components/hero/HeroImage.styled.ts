import styled from 'styled-components'
import tw from 'twin.macro'

export const HeroWrapper = styled.div`
  ${tw`relative flex mb-4 items-stretch`}
`

export const HeroInner = styled.div`
  ${tw`relative w-full font-display flex-none sm:flex-1 md:flex-auto lg:flex-initial xl:flex-1`}

  & > img {
    ${tw`w-full`}
  }
`

export const HeroHeading = styled.h1`
  ${tw`text-white w-48 md:w-auto leading-4 md:leading-5 xl:leading-10 font-bold 
       text-xs sm:text-sm md:text-xl lg:text-2xl xl:text-5xl uppercase mb-2 sm:mb-4`}

`

export const HeroDescription = styled.p`
  ${tw`mb-2 w-48 md:w-auto leading-3 md:leading-5 font-bold sm:mb-8 text-xs md:text-base lg:text-lg xl:text-xl`}
`

export const BannerText = styled.div`
  ${tw`text-white absolute flex flex-col justify-around`}
  top: 25%;
  left: 20%;

  @media(min-width: 1024px){
    top: 35%;
  }

  & > button {
    ${tw`w-1/2 lg:w-2/5 py-2 md:text-base`}
    font-size: 0.5rem;
  }
`
