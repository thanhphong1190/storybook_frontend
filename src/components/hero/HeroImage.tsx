import { Button } from '../base'
import {
  HeroWrapper,
  HeroInner,
  BannerText,
  HeroHeading,
  HeroDescription
} from './HeroImage.styled'

export interface HeroImageProps {
  heading: string
  description: string
  imgSrc: string
  joinBtnText: string
  onTryBtnClick?: Function
}

export const HeroImage = (props: HeroImageProps) => {
  const { heading, description, imgSrc, joinBtnText, onTryBtnClick } = props

  return (
    <HeroWrapper>
      <HeroInner>
        <BannerText>
          <HeroHeading dangerouslySetInnerHTML={{ __html: heading }} />
          <HeroDescription dangerouslySetInnerHTML={{ __html: description }} />
          <Button onClick={() => onTryBtnClick && onTryBtnClick()}>
            {joinBtnText}
          </Button>
        </BannerText>
        <img src={imgSrc} />
      </HeroInner>
    </HeroWrapper>
  )
}

export default HeroImage
