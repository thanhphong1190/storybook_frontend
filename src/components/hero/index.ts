export { Hero, HeroProps } from './Hero'
export { HeroImage, HeroImageProps } from './HeroImage'
