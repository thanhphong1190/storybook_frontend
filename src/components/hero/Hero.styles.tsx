import styled from 'styled-components'
import tw from 'twin.macro'
import { Text, ButtonLink } from '../../styles/global.styles'
import { HerorTextProps } from './Hero'

export const Section = styled.section<{ bgImgURL?: string; bgColor?: string }>`
         ${tw`w-full bg-center bg-cover bg-no-repeat`}
         ${({ bgColor }) => {
           switch (bgColor) {
             case 'orange':
               return tw`bg-orange`
             case 'blue':
               return tw`bg-blue`
             case 'dark-blue':
               return tw`bg-blue__dark`
             case 'dodger-blue':
               return tw`bg-blue__dodger`
             case 'white':
               return tw`bg-white`
             case 'grey':
               return tw`bg-grey`
             default:
               return null
           }
         }}
         &{
             ${({ bgImgURL }) =>
               bgImgURL &&
               `background-image:url('${bgImgURL}'); ${tw`bg-center bg-cover`}`}
       `
export const OverlayWrapper = styled.div<{ overlay?: boolean }>`
  ${tw`w-full`}
`
export const Container = styled.div<{
  contentPosition?: 'center' | 'left' | 'right'
}>`
  ${tw`container mx-auto flex flex-col`}

  ${({ contentPosition }) => {
    switch (contentPosition) {
      case 'center':
        return tw`justify-center text-center`
      case 'left':
        return tw`justify-start text-left`
      case 'right':
        return tw`justify-end text-right`
      default:
        return tw`justify-center text-center`
    }
  }}
`

export const ButtonGroup = styled.div<{
  contentPosition?: 'center' | 'left' | 'right'
}>`
  ${tw`py-2 mt-4 flex flex-wrap flex-row items-center`}
  ${({ contentPosition }) => {
    switch (contentPosition) {
      case 'center':
        return tw`justify-center text-center`
      case 'left':
        return tw`justify-start text-left`
      case 'right':
        return tw`justify-end text-right`
      default:
        return tw`justify-center text-center`
    }
  }}
`

export const HeroText = styled(Text)<HerorTextProps>`
  ${tw`lg:leading-none`}
  ${({ size }) => {
    switch (size) {
      case '4xl':
        return tw`leading-8`
      case '2xl':
        return tw`leading-6`
    }
  }}
  ${({ spaceTop }) => spaceTop && `margin-top:${spaceTop}em;`}
  ${({ spaceBottom }) => spaceBottom && `margin-bottom:${spaceBottom}em;`}
`

export const HeroButtonLink = styled(ButtonLink)`
  ${tw`mr-4 my-2 py-4 px-4 lg:px-8`}
`
export const ContentWrapper = styled.div<{
  width?: 'full' | '1/2' | '2/3' | '3/4' | '4/5'
}>`
  ${tw`w-full px-2 py-24 lg:px-4 lg:pb-40 lg:pt-64`}
  ${({ width }) => {
    switch (width) {
      case '1/2':
        return tw`md:w-2/3 lg:w-1/2`
      case '2/3':
        return tw`lg:w-2/3`
      case '3/4':
        return tw`lg:w-3/4`
      case '4/5':
        return tw`lg:w-4/5`
      case 'full':
        return tw`lg:w-full`
      default:
        return null
    }
  }}
`
