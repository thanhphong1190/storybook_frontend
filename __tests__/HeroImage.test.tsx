import { HeroImage, HeroImageProps } from '../src/components/hero/HeroImage'
import renderer from 'react-test-renderer'

const props: HeroImageProps = {
  heading: `FREE 3 DAY TRIAL <br /> It's gym time`,
  description: `Come and check us out.<br />We know youâ€™ll like what you see!`,
  imgSrc: `/example/home-hero.png`,
  joinBtnText: 'TRY NOW!',
}

it('renders correctly', () => {
  const tree = renderer.create(<HeroImage {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
