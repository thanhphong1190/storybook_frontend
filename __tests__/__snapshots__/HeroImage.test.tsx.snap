// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="HeroImagestyled__HeroWrapper-vzwf23-0 GfeIL"
>
  <div
    className="HeroImagestyled__HeroInner-vzwf23-1 crHRFq"
  >
    <div
      className="HeroImagestyled__BannerText-vzwf23-4 fdrzsk"
    >
      <h1
        className="HeroImagestyled__HeroHeading-vzwf23-2 UXJco"
        dangerouslySetInnerHTML={
          Object {
            "__html": "FREE 3 DAY TRIAL <br /> It's gym time",
          }
        }
      />
      <p
        className="HeroImagestyled__HeroDescription-vzwf23-3 hXTBCN"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Come and check us out.<br />We know youâ€™ll like what you see!",
          }
        }
      />
      <button
        className="Button-lxhj0a-0 eQpnsG"
        onClick={[Function]}
      >
        TRY NOW!
      </button>
    </div>
    <img
      src="/example/home-hero.png"
    />
  </div>
</div>
`;
